### PR TITLE
Reduce handle inputs and remove update state from tools

### PR DESCRIPTION
**Pull request by guerler**
The handle input function in tools used to play a vital role in rendering and refreshing tool states. Now it is used for tool execution only. This allows us to simplify it and remove unused parts from the tools class: https://github.com/galaxyproject/galaxy/pull/829/files#diff-1bef786a2b2ea704c1748e41cd25dd33R45. Some unused upload related tool runner actions and JS sections have also been removed: https://github.com/galaxyproject/galaxy/pull/829/files#diff-99430eab480b0bc21518d808dff6ccdfL181. By removing the concept of mako templates from tools, the error handling can now consistently raise and try common exceptions instead of parsing custom dictionaries e.g. for the message.mako: https://github.com/galaxyproject/galaxy/pull/829/files#diff-1bef786a2b2ea704c1748e41cd25dd33L1363. The tool form displays job submission exceptions properly now. Overall this PR is following up on: https://github.com/galaxyproject/galaxy/pull/786.

### CODE REVIEW

**Comment by jmchilton** [comment]
Looks like the tests caught one real error:

```
======================================================================
ERROR: test_multirun_with_state_updates (tools.test_execution.ToolExecutionTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/galaxyproject/galaxy/test/unit/tools/test_execution.py", line 248, in test_multirun_with_state_updates
    template, template_vars = self.__handle_with_incoming()
  File "/home/travis/build/galaxyproject/galaxy/test/unit/tools/test_execution.py", line 366, in __handle_with_incoming
    incoming=incoming,
  File "/home/travis/build/galaxyproject/galaxy/lib/galaxy/tools/__init__.py", line 1330, in handle_input
    template_vars = dict( errors=errors, tool_state=state, incoming=incoming, error_message=error_message )
NameError: global name 'state' is not defined
```

The rest of the failing tests were testing state update logic which we don't do anymore. Ideally they would be rewritten to use to_json or some other API/functional test would cover the same functionality. The old code was hard to follow and messy but it did have some tests. I don't think that needs to stop merging of this - if you just delete the tests we can merge this, but I did need to register some distress at loosing test coverage on this very important part of the Galaxy code base.

**Comment by martenson** [comment]
both data library occurrences seem to work fine with the switch to `populate_state`

**Comment by guerler** [comment]
Ok, I'll go through the tests and yes I agree, we have to replicate the tests for the JS part.

**Comment by martenson** [comment]
@galaxybot test this please

**Comment by guerler** [comment]
Thx.

**Comment by martenson** [comment]
The API tests for this have multiple fails, details: https://jenkins.galaxyproject.org/job/docker-api/318/testReport/

**Comment by nsoranzo** [comment]
@martenson Would it be better to open an issue since this has already been merged?

**Comment by martenson** [comment]
@nsoranzo I notified @guerler via IRC and he is on it, this is just for reference

**Comment by martenson** [comment]
being handled here: https://github.com/galaxyproject/galaxy/pull/876
